src/Microsoft.Crank.Agent/Startup.cs

```diff
@@ -2187,16 +2187,47 @@ private static async Task<(string containerId, string imageName, string workingD
                 environmentArguments += $"--memory=\"{job.MemoryLimitInBytes}b\" ";
             }
 
-            var command = $"run -d {environmentArguments} {job.Arguments} --label benchmarks --name {containerName} --privileged --network host {imageName} {job.DockerCommand}";
+            // docker create --name {containerName}
+            var createCommand = $"create {environmentArguments} {job.Arguments} --label benchmarks --name {containerName} --privileged --network host {imageName} {job.DockerCommand}";
+
+            job.BuildLog.AddLine("docker " + createCommand);
+
+            var createCommandResult = await ProcessUtil.RunAsync("docker", $"{createCommand} ",
+                throwOnError: true,
+                onStart: _ => stopwatch.Start(),
+                captureOutput: true,
+                log: true,
+                outputDataReceived: job.BuildLog.AddLine
+            );
+
+            // Copy attachments to container.
+            foreach (var attachment in job.Attachments)
+            {
+                var filename = attachment.Filename.Replace("\\", "/");
+                var tempFilePath = attachment.TempFilename;
+
+                Log.Info($"Copying output file to container: {filename}");
+
+                string dockerCopyCommand = $"cp {tempFilePath} {containerName}:{filename}";
+                var copyResult = await ProcessUtil.RunAsync("docker", $"{dockerCopyCommand} ",
+                    throwOnError: true,
```

Comment:
Should we worry about stale container created if this fails?

Reply:
Yes, definitely - guarding against that now.